### PR TITLE
[UI] Correct indicators for context switcher.

### LINKED ISSUE #9641: Kubernetes Connection Status Misreprsented
https://github.com/meshery/meshery/assets/7570704/832b5b78-fb05-4f51-b4c8-84cb5c7d9c52

### PR DESCRIPTION
**Pull request by MUzairS15**
**Notes for Reviewers**

This PR fixes #9641 

1. Modularizes connection chip.
2. Correct light indicators in context switcher.
3. Correct the API endpoint being called when clicking on `x` icon in the connection chip.
4. Use ConnectioChip in the context switcher.

https://github.com/meshery/meshery/assets/79860504/a4a8f8cd-dfb1-434a-8dfb-b9ed2db3548e

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by MUzairS15** [comment]
// @leecalcote

**Comment by leecalcote** [comment]
Thanks for this screen recording.

**Comment by MUzairS15** [comment]
Merging by EOD.

File: ui/components/DashboardComponent/resources/configuration/config.js [hunk 1]
```diff
@@ -113,7 +113,10 @@ export const ConfigurationTableConfig = (switchView, meshSyncResources, k8sConfi
               let clusterName = getClusterNameFromClusterId(val, k8sConfig);
               return (
                 <>
-                  <ConnectionChip title={clusterName} iconSrc="/static/img/kubernetes.svg" />
+                  <TootltipWrappedConnectionChip
+                    title={clusterName}
+                    iconSrc="/static/img/kubernetes.svg"
```
Comment (leecalcote) [inline ui/components/DashboardComponent/resources/configuration/config.js:118]: Should this icon source be a variable reference?

File: ui/components/DashboardComponent/resources/network/config.js [hunk 2]
```diff
@@ -227,7 +227,7 @@ export const NetWorkTableConfig = (switchView, meshSyncResources, k8sConfig) =>
               let connectionId = getConnectionIdFromClusterId(val, k8sConfig);
               return (
                 <>
-                  <ConnectionChip
+                  <TootltipWrappedConnectionChip
                     title={clusterName}
                     iconSrc="/static/img/kubernetes.svg"
```
Comment (leecalcote) [inline ui/components/DashboardComponent/resources/network/config.js:232]: Same question here.

File: ui/components/DashboardComponent/resources/nodes/config.js [hunk 3]
```diff
@@ -129,7 +129,7 @@ export const NodeTableConfig = (switchView, meshSyncResources, k8sConfig) => {
             let connectionId = getConnectionIdFromClusterId(val, k8sConfig);
             return (
               <>
-                <ConnectionChip
+                <TootltipWrappedConnectionChip
                   title={clusterName}
                   iconSrc="/static/img/kubernetes.svg"
```
Comment (leecalcote) [inline ui/components/DashboardComponent/resources/nodes/config.js:134]: And here.

File: ui/components/DashboardComponent/resources/security/config.js [hunk 4]
```diff
@@ -231,7 +231,7 @@ export const SecurityTypesConfig = (switchView, meshSyncResources, k8sConfig) =>
               let connectionId = getConnectionIdFromClusterId(val, k8sConfig);
               return (
                 <>
-                  <ConnectionChip
+                  <TootltipWrappedConnectionChip
                     title={clusterName}
                     iconSrc="/static/img/kubernetes.svg"
```
Comment (leecalcote) [inline ui/components/DashboardComponent/resources/security/config.js:236]: Hmm

File: ui/components/DashboardComponent/resources/storage/config.js [hunk 5]
```diff
@@ -142,7 +142,7 @@ export const StorageTableConfig = (switchView, meshSyncResources, k8sConfig) =>
               let connectionId = getConnectionIdFromClusterId(val, k8sConfig);
               return (
                 <>
-                  <ConnectionChip
+                  <TootltipWrappedConnectionChip
                     title={clusterName}
                     iconSrc="/static/img/kubernetes.svg"
```
Comment (leecalcote) [inline ui/components/DashboardComponent/resources/storage/config.js:147]: Goodness

File: ui/components/DashboardComponent/resources/workloads/config.js [hunk 6]
```diff
@@ -188,7 +188,7 @@ export const WorkloadTableConfig = (switchView, meshSyncResources, k8sConfig) =>
               let connectionId = getConnectionIdFromClusterId(val, k8sConfig);
               return (
                 <>
-                  <ConnectionChip
+                  <TootltipWrappedConnectionChip
                     title={clusterName}
                     iconSrc="/static/img/kubernetes.svg"
```
Comment (leecalcote) [inline ui/components/DashboardComponent/resources/workloads/config.js:193]: The amount of hardcoding…

File: ui/components/DashboardComponent/resources/workloads/config.js [hunk 7]
```diff
@@ -355,7 +355,7 @@ export const WorkloadTableConfig = (switchView, meshSyncResources, k8sConfig) =>
...
```
Comment (leecalcote) [inline ui/components/DashboardComponent/resources/workloads/config.js:360]: Oof

File: ui/components/Header.js [hunk 8]
```diff
@@ -493,30 +409,12 @@ function K8sContextMenu({
...
-                                  className={classes.icon}
-                                  style={operStatus ? {} : { opacity: 0.2 }}
-                                />
-                              )
-                            }
-                            variant="filled"
-                            className={classes.Chip}
-                            data-cy="chipContextName"
+                            handlePing={() => ping(ctx.name, ctx.connection_id, ctx.name)}
+                            iconSrc="/static/img/kubernetes.svg" // chnage to use connection def
```
Comment (leecalcote) [inline ui/components/Header.js:416]: Good note in this comment. Let’s do it.

File: ui/components/connections/ConnectionChip.js [hunk 9]
```diff
@@ -33,41 +33,46 @@ const useChipStyles = makeStyles(() => ({
...
+    // <Tooltip title={tooltip || title} placement="bottom">
+    <Chip
+      label={title}
+      onClick={handlePing}
+      onDelete={onDelete}
+      avatar={
+        status ? (
+          <BadgeAvatars
+            color={
+              status === CONNECTION_STATES.CONNECTED || status === CONTROLLER_STATES.DEPLOYED
```
Comment (leecalcote) [inline ui/components/connections/ConnectionChip.js:48]: @MUzairS15 we would be better off if this logic was centralized in the form of the pairing of central constants.

**Comment by MUzairS15** [comment]
Added review changes.

**Comment by leecalcote** [comment]
> Added review changes.

Most excellent. 😀

**Pull request merged**